app/views/widgets/_smart_banner.html.slim

```diff
@@ -0,0 +1,13 @@
+- cookie = cookies[:smart_banner_dismiss]
+- if (browser.device.mobile? || browser.device.tablet?) && (cookie.nil? || (Time.now >= Time.strptime(cookie, "%FT%R") + 15.days))
```

Comment:
I suggest you to move all this logic to a view helper, so your view code stays clean:
```
- if allowed_to_show_smart_banner? // Implement this on your view helper
  .smart-banner
     ...
```